'devices(for:)' was deprecated in macOS

I found this when trying to launch in macOS 

```
.pub-cache/hosted/pub.dev/mobile_scanner-7.1.2/darwin/mobile_scanner/Sources/mobile_scanner/MobileScannerPlugin.swift:384:38: warning: 'devices(for:)' was deprecated in macOS 10.15: Use AVCaptureDeviceDiscoverySession instead.
            device = AVCaptureDevice.devices(for: .video).filter({$0.position == position}).first
                                     ^
warning: Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this issue, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Flutter Assemble' from project 'Runner')
```

--- a/darwin/mobile_scanner/Sources/mobile_scanner/MobileScannerCameraSelector.swift
+++ b/darwin/mobile_scanner/Sources/mobile_scanner/MobileScannerCameraSelector.swift
@@ -97,10 +97,14 @@ class MobileScannerCameraSelector {
             return nil
         }
 
-        // Legacy fallback for older OS versions: filter by position
+#if os(iOS)
+        // Legacy fallback for iOS < 13.0: filter by position.
+        // On macOS, DiscoverySession handles 10.15+; older macOS falls
+        // through to AVCaptureDevice.default(for:) below.
         if let device = AVCaptureDevice.devices(for: .video).filter({ $0.position == position }).first {
             return device
         }
+#endif
 
         // Ultimate fallback: any available video device
         return AVCaptureDevice.default(for: .video)

